controllers/certificatepolicy_controller.go

```diff
@@ -173,118 +136,67 @@ func (r *CertificatePolicyReconciler) PeriodicallyExecCertificatePolicies(
 
 // ProcessPolicies reads each policy and looks for violations returning true if a change is found.
 func (r *CertificatePolicyReconciler) ProcessPolicies(
-	ctx context.Context, plcToUpdateMap map[string]*policyv1.CertificatePolicy,
-) bool {
-	stateChange := false
+	ctx context.Context, policies *policyv1.CertificatePolicyList,
+) []*policyv1.CertificatePolicy {
+	updatedPolicies := map[types.NamespacedName]*policyv1.CertificatePolicy{}
 
-	plcMap := make(map[string]*policyv1.CertificatePolicy)
-	// create a map of all policies
-	for _, policy := range availablePolicies.PolicyMap {
-		plcMap[policy.Name] = policy
-	}
 	// update available policies if there are changed namespaces
-	for _, plc := range plcMap {
-		// Retrieve the namespaces based on filters in NamespaceSelector
-		selectedNamespaces := r.retrieveNamespaces(ctx, plc.Spec.NamespaceSelector)
-
-		// add availablePolicy if not present
-		for _, ns := range selectedNamespaces {
-			key := fmt.Sprintf("%s/%s", ns, plc.Name)
-			_, found := availablePolicies.GetObject(key)
-
-			if !found {
-				availablePolicies.AddObject(key, plc)
-				plcToUpdateMap[plc.Name] = plc
-				// remove the dummy entry not matching namespaces if it exists
-				cleanupAvailablePolicies("", plc.Name)
-			}
-		}
+	for i := range policies.Items {
+		policy := policies.Items[i]
 
-		handleNamespaceRemovals(plc, plcToUpdateMap, selectedNamespaces)
+		namespacedName := types.NamespacedName{Namespace: policy.Namespace, Name: policy.Name}
 
-		if len(selectedNamespaces) == 0 {
-			// add a dummy entry to force updates when no namespaces match
-			key := fmt.Sprintf("/%s", plc.Name)
-			_, found := availablePolicies.GetObject(key)
+		log.Info("Checking certificates", "policy.Name", policy.Name)
 
-			if !found {
-				availablePolicies.AddObject(key, plc)
-				plcToUpdateMap[plc.Name] = plc
-			}
+		// Retrieve the namespaces based on filters in NamespaceSelector
+		selectedNamespaces := r.retrieveNamespaces(ctx, policy.Spec.NamespaceSelector)
+
+		// If there are no applicable namespaces, make the policy compliant
+		if len(selectedNamespaces) == 0 {
```

Comment:
Why this?
`retrieveNamespaces` return `[]string`

Reply:
The existing logic of `buildPolicyStatusMessage` handles the case where no namespaces were selected (e.g. no namespaces have a particular label) and builds the message for this case when the namespace is equal to `""`.